### PR TITLE
Clarify convention for plugin class naming.

### PR DESCRIPTION
**Pull request by dereuromark**
As the current wording is not clear.

### CODE REVIEW

File: en/appendices/4-4-migration-guide.rst [hunk 1]
```diff
@@ -112,10 +112,11 @@ ORM
 Plugins
 -------
 
-- Plugin class names now match the plugin name with a "Plugin" suffix. For
-  example, the plugin class for "ADmad/I18n" would be ``ADmad\I18n\I18nPlugin``
+- Plugin class names should now match the plugin name with a "Plugin" suffix. For
+  example, the plugin class for ``ADmad/I18n`` plugin would be ``ADmad\I18n\I18nPlugin``
   instead of ``ADmad\I18n\Plugin``, as was the case for CakePHP 4.3 and below.
-  The old style name is still supported for backwards compatibility.
+  The old style name for existing majors should be kept to avoid BC breaks.
```
Comment (ravage84) [inline en/appendices/4-4-migration-guide.rst]: I think this could be clarified even further.

File: en/appendices/4-4-migration-guide.rst [hunk 2]
```diff
@@ -112,10 +112,11 @@ ORM
...
 -------
 
-- Plugin class names now match the plugin name with a "Plugin" suffix. For
-  example, the plugin class for "ADmad/I18n" would be ``ADmad\I18n\I18nPlugin``
+- Plugin class names should now match the plugin name with a "Plugin" suffix. For
+  example, the plugin class for ``ADmad/I18n`` plugin would be ``ADmad\I18n\I18nPlugin``
   instead of ``ADmad\I18n\Plugin``, as was the case for CakePHP 4.3 and below.
-  The old style name is still supported for backwards compatibility.
+  The old style name for existing majors should be kept to avoid BC breaks. 
+  So the new convention is for new plugins or when doing a new major release.
```
Comment (ravage84) [inline en/appendices/4-4-migration-guide.rst]: Style-wise, I think, a sentense shouldn't start with "So". Especially not in written English. But I'm not a native speaker.

Comment (ravage84) [reply en/appendices/4-4-migration-guide.rst]:
```suggestion
  The new plugin classs naming convention should be folowed when developing a new plugin or
  when doing a new major release for an existing plugin.
```

Comment (dereuromark) [reply en/appendices/4-4-migration-guide.rst]: typo `classs` and `folowed`.
Otherwise fine by me.

**Review by Arhell — APPROVED** [review]

**Pull request merged**